app/controllers/events_controller.rb

```diff
@@ -132,6 +144,33 @@ def start_review
       redirect_to event_event_rating_path(event_id: ids.first)
     end
   end
+  
+  # batch actions
+  def batch_actions
+    if params[:bulk_email]
+      bulk_send_email
+    else
+      redirect_to events_path, alert: :illegal
+    end
+  end
+  
+  def bulk_send_email
+    authorize @conference, :orga?
+    
+    mail_template = @conference.mail_templates.find_by(name: params[:template_name])
+    redirect_back(alert: t('ability.denied'), fallback_location: root_path) and return if mail_template.blank?
+    
+    events = search @conference.events_with_review_averages.includes(:track)
+    event_people = EventPerson.where(event_id: events.to_a.pluck(:id))
+    
+    if Rails.env.production?
+      SendBulkMailJob.new.async.perform(mail_template, event_people)
+      redirect_back(notice: t('emails_module.notice_mails_queued'), fallback_location: root_path)
+    else
+      SendBulkMailJob.new.perform(mail_template, event_people)
```

Comment:
Why would development use the more robust approach of synchronous delivery? I would hate seeing Bulk Mail working in dev mode and then wonder why this doesn't work anymore with frab users in the wild …

Isn't that something that should be configurable independently of production/development settings?

Reply:
I copied from https://github.com/frab/frab/commit/8e2404c7eb9c11063995ea6c4657f467fded87fb#diff-2f9d0e8ee584f04039c864ea0da162daR14-R22 . It was there for 5 years.

I think the reason they do synchronously for deelopment is that it will let it open the shim e-mail as a new browser window.

Reply:
Ah, I see, thanks. So I hope someone did think this through in the past. ;)